packages/errors/src/index.ts

```diff
@@ -0,0 +1,122 @@
+  error.code === 'ConditionalCheckFailedException';
+
+/**
+ * WorkflowError should be bubbled out to the overall workflow in the 'exception' field, rather than
+ * being thrown and causting an immediate failure
+ */
+export const WorkflowError = createErrorType('WorkflowError');
+
+/**
+ * NotNeededError indicates that execution was not completed because it was unnecessary. The
+ * workflow should therefore terminate but be considered successful
+ */
+export const NotNeededError = createErrorType('NotNeededWorkflowError', WorkflowError);
+
+/**
+ * IncompleteError indicates that the execution was partially successful and can be re-executed to
+ * make further progress. This may happen, for instance, if an execution timeout stops progress
+ */
+export const IncompleteError = createErrorType('IncompleteWorkflowError', WorkflowError);
+
+/**
+ * ResourcesLockedError indicates that the execution is unable to proceed due to resources being
+ * tied up in other executions. Execution may be retried after resources free up
+ */
+export const ResourcesLockedError = createErrorType('ResourcesLockedWorkflowError', WorkflowError);
+
+/**
+ * RemoteResourceError indicates that a required remote resource could not be fetched or otherwise
+ * used
+ */
+export const RemoteResourceError = createErrorType('RemoteResourceError');
+
+/**
+ * The error object for when the xml file path is not provided
+ */
+export const XmlMetaFileNotFound = createErrorType('XmlMetaFileNotFound');
+
+/**
+ * No CMR metadata file was present.
+ */
+export const CMRMetaFileNotFound = createErrorType('CMRMetaFileNotFound');
+
+/**
+ * The provider info is missing error
+ */
+export const ProviderNotFound = createErrorType('ProviderNotFound');
+
+export const FTPError = createErrorType('FTPError');
+
+export const PDRParsingError = createErrorType('PDRParsingError');
+
+export const ConnectionTimeout = createErrorType('ConnectionTimeout');
+
+export const HostNotFound = createErrorType('HostNotFound');
+
+export const FileNotFound = createErrorType('FileNotFound');
+
+export const InvalidChecksum = createErrorType('InvalidChecksum');
+
+export const DuplicateFile = createErrorType('DuplicateFile');
+
+export const UnexpectedFileSize = createErrorType('UnexpectedFileSize');
+
+/**
+ * Error thrown when system encounters a conflicting request.
+ */
+export const InvalidArgument = createErrorType('InvalidArgument');
+
+/**
+ * is raised if the PDR file doesn't match the collection
+ */
+export const MismatchPdrCollection = createErrorType('MismatchPdrCollection');
+
+export const UnparsableFileLocationError = createErrorType('UnparsableFileLocationError');
+
+export const RecordDoesNotExist = createErrorType('RecordDoesNotExist');
+
+export const InvalidRegexError = createErrorType('InvalidRegexError');
+
+export const UnmatchedRegexError = createErrorType('UnmatchedRegexError');
+
+export const ValidationError = createErrorType('ValidationError');
```

Comment:
So it doesn't matter if we write our TS using export/import syntax because it compiles to JS using CommonJS style? I actually prefer export/import syntax, but is confusing to have different styles in different places?

Reply:
It actually does matter, but in a good way. Typescript uses the export / import syntax only. It will then compile it to the module format that you specify in your `tsconfig.json` file.